src/modules/ir/ir_read.cpp

```diff
@@ -64,19 +90,35 @@ void IrRead::setup() {
     pinMode(bruceConfig.irRx, INPUT);
     if(headless) return;
     // else
-    begin();
-    return loop();
+    returnToMenu = true;  // make sure menu is redrawn when quitting in any point
+    options = {
+        {"Custom Read", [&]() { begin(); return loop(); }},
+        {"Quick Remote Setup  ", [&]() { quickloop = true; begin(); return loop();}},
+        {"Menu", []() { }},
+    };
+    loopOptions(options);
+    
 }
 
+
 void IrRead::loop() {
     while(1) {
         if (check(EscPress)) {
             returnToMenu=true;
+            button_pos = 0;
+            quickloop = false;
             break;
         }
-
+        if (quickloop) {
+            if (check(NextPress)) save_signal(quickButtons[button_pos]);
+        }
+        else {
+            if (check(NextPress)) save_signal("null");
+        }
```

Comment:
```suggestion
        if (check(NextPress)) save_signal();
```
And why giving "null" as a parameter, just check for an empty string.